e2e_tests/tests/experiment/test_core.py

```diff
@@ -476,3 +476,25 @@ def test_max_concurrent_trials(name: str, searcher_cfg: str) -> None:
 
     finally:
         exp.cancel_single(experiment_id)
+
+
+@pytest.mark.e2e_cpu
+def test_core_api_non_conforming_steps_vals_checkpoints() -> None:
```

Comment:
I suggest renaming "non-conforming" to "arbitrary order" or something.

Because after people get used to the idea that metrics and validations are just arbitrary time-value series reported in arbitrary orders, the name "non conforming" isn't going to have any meaning.